lib/ecto_ldap/adapter/sandbox.ex

```diff
@@ -35,34 +39,70 @@ defmodule Ecto.Ldap.Adapter.Sandbox do
-    {:ok, {:eldap_search_result, [@jeffweiss], []}}
+  def handle_call({:search, %{scope: :baseObject, base: 'uid=manny,ou=users,dc=example,dc=com'}}, _from, state) do
+    ldap_response = {:ok, {:eldap_search_result, [List.last(state)], []}}
+    {:reply, ldap_response, state}
   end
-  def search(_pid, %{scope: :baseObject, base: 'uid=manny,ou=users,dc=example,dc=com'}) do
-    {:ok, {:eldap_search_result, [@manny], []}}
+  def handle_call({:search, %{scope: :baseObject}}, _from, state) do
+    ldap_response = {:ok, {:eldap_search_result, [], []}}
+    {:reply, ldap_response, state}
   end
-  def search(_pid, %{scope: :baseObject}) do
-    {:ok, {:eldap_search_result, [], []}}
+  def handle_call({:search, %{base: 'ou=users,dc=example,dc=com', filter: {:and, [and: [equalityMatch: {:AttributeValueAssertion, 'uid', 'jeff.weiss'}], and: []]}}}, _from, state) do
+    ldap_response = {:ok, {:eldap_search_result, [List.first(state)], []}}
+    {:reply, ldap_response, state}
   end
-  def search(_pid, %{base: 'ou=users,dc=example,dc=com', filter: {:and, [and: [equalityMatch: {:AttributeValueAssertion, 'uid', 'jeff.weiss'}], and: []]}}) do
-    {:ok, {:eldap_search_result, [@jeffweiss], []}}
+  def handle_call({:search, %{base: 'ou=users,dc=example,dc=com', filter: {:and, [and: [], and: [equalityMatch: {:AttributeValueAssertion, 'uid', 'jeff.weiss'}]]}}}, _from, state) do
+    ldap_response = {:ok, {:eldap_search_result, [List.first(state)], []}}
+    {:reply, ldap_response, state}
   end
-  def search(_pid, %{base: 'ou=users,dc=example,dc=com', filter: {:and, [and: [], and: [equalityMatch: {:AttributeValueAssertion, 'uid', 'jeff.weiss'}]]}}) do
-    {:ok, {:eldap_search_result, [@jeffweiss], []}}
+  def handle_call({:search, %{base: 'ou=users,dc=example,dc=com'}}, _from, state) do
+    ldap_response = {:ok, {:eldap_search_result, state, []}}
+    {:reply, ldap_response, state}
   end
-  def search(_pid, %{base: 'ou=users,dc=example,dc=com'}) do
-    {:ok, {:eldap_search_result, [@jeffweiss, @manny], []}}
+  def handle_call({:search, _search_options}, _from, state) do
+    ldap_response = {:ok, {:eldap_search_result, [], []}}
+    {:reply, ldap_response, state}
   end
-  def search(_pid, _search_options) do
-    {:ok, {:eldap_search_result, [], []}}
+  def handle_call({:update, 'uid=manny,ou=users,dc=example,dc=com', modify_operations}, _from, state) do
+    {:eldap_entry, dn, attributes} = List.last(state)
+
+    attribute_map       = Enum.into(attributes, %{})
+    updated_attributes  = Enum.reduce(
+      modify_operations,
+      attribute_map,
+      fn ({:ModifyRequest_changes_SEQOF, :replace, {:PartialAttribute, attribute, []}}, attribute_map) ->
+          Map.update!(attribute_map, attribute, fn _ -> nil end)
+         ({:ModifyRequest_changes_SEQOF, :replace, {:PartialAttribute, attribute, new_value}}, attribute_map) ->
+          Map.update!(attribute_map, attribute, fn _ -> new_value end) end)
```

Comment:
The readability of this section is problematic because we have `attribute_map` within the reduce function shadowing the binding of `attribute_map` of the overall function.